src/snapshots.js

```diff
@@ -188,38 +188,54 @@ export async function* takeStorybookSnapshots(percy, callback, { baseUrl, flags
     // set storybook environment info
     percy.client.addEnvironmentInfo(environmentInfo);
 
-    // use a single page to capture story snapshots without reloading
-    yield* withPage(percy, previewUrl, async function*(page) {
-      // determines when to retry page crashes
-      lastCount = snapshots.length;
-
-      while (snapshots.length) {
-        // separate story and snapshot options
-        let { id, args, globals, queryParams, ...options } = snapshots[0];
-
-        if (flags.dryRun || options.enableJavaScript || percy.config.snapshot.enableJavaScript) {
-          // when dry-running or when javascript is enabled, use the preview dom
-          options.domSnapshot = previewResource.content;
+    while (snapshots.length) {
```

Comment:
nit: add a comment here why we're not reusing the same page.

Reply:
Added